debvader/debvader.py

```diff
@@ -209,25 +210,15 @@ def iterative_deblending(net, field_image, galaxy_distances_to_center_in, npeaks
     denoised_field_epistemic_total = denoised_field_epistemic
     cutout_images_total = cutout_images
     output_images_total = output_images_mean
-    galaxy_distances_to_center_total = galaxy_distances_to_center
-    detection_k=np.zeros((1))
+    nb_of_galaxies_in_deblended_field_total = nb_of_galaxies_in_deblended_field
 
-    while (len(shifts)>len(shifts_previous)):#len(detection_k)!=0:#
+    while (len(shifts)>len(shifts_previous)):
 
         print("iteration "+str(k))
-        #print(npeaks_per_iteration)
         mse_step_previous=mse_step
         shifts_previous = shifts
-        detection_k = detect_objects(field_image, npeaks_per_iteration=npeaks_per_iteration, factor=factor)
-        # Avoid to have several detection at the same location
-        idx_to_remove = []
-        for i in range (len(detection_k)):
-            if detection_k[i] in galaxy_distances_to_center_total:
-                idx_to_remove.append(i)
-        detection_k = np.delete(detection_k, idx_to_remove, axis = 0)
-        detection_up_to_k = np.concatenate((detection_k,galaxy_distances_to_center_total),axis=0)
 
-        field_img_save, field_image, denoised_field, denoised_field_std, denoised_field_epistemic, cutout_images, output_images_mean, output_images_distribution, shifts, galaxy_distances_to_center, mse_step = deblending_step(net, field_image, detection_k, cutout_images = None, cutout_size = cutout_size, nb_of_bands = nb_of_bands, optimise_positions=optimise_positions, epistemic_uncertainty_estimation=epistemic_uncertainty_estimation, epistemic_criterion=epistemic_criterion, mse_criterion=mse_criterion, normalised=normalised)
+        field_img_save, field_image, denoised_field, denoised_field_std, denoised_field_epistemic, cutout_images, output_images_mean, output_images_distribution, shifts, galaxy_distances_to_center_total, mse_step, nb_of_galaxies_in_deblended_field = deblending_step(net, field_image, galaxy_distances_to_center_total, cutout_images = None, cutout_size = cutout_size, nb_of_bands = nb_of_bands, optimise_positions=optimise_positions, epistemic_uncertainty_estimation=epistemic_uncertainty_estimation, epistemic_criterion=epistemic_criterion, mse_criterion=mse_criterion, normalised=normalised,factor=factor)
```

Comment:
here again, way too complicated, and too easy to make a mistake

Reply:
Same, see [7754ad3](https://github.com/BastienArcelin/debvader/commit/7754ad39feb606cfee8e4b0f776da5645eb16258#diff-a5dae2bb532104d1ce10a8b52d5c651b023b7d0d57e87a12c0c6942e0d44cee7)